### PR TITLE
Add formulas for distributions

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #69956
* #69884
* #69727
* #69740
* __->__ #69690
* #69661

Differential Revision: [D33031726](https://our.internmc.facebook.com/intern/diff/D33031726)

### CODE REVIEW

**Review by albanD — APPROVED** [review]
We want to make these non-differentiable in the future. So sure.

File: tools/autograd/derivatives.yaml [hunk 1]
```diff
@@ -1822,15 +1823,19 @@
...
   self: softplus_backward(grad, self, beta, threshold, result)
+  result: auto_element_wise
 
 - name: softshrink(Tensor self, Scalar lambd=0.5) -> Tensor
   self: softshrink_backward(grad, self, lambd)
+  result: auto_element_wise
 
 - name: threshold(Tensor self, Scalar threshold, Scalar value) -> Tensor
   self: threshold_backward(grad, self, threshold)
+  result: auto_element_wise
```
Comment (albanD) [inline tools/autograd/derivatives.yaml]: why did this got moved here?

Comment (soulitzer) [reply tools/autograd/derivatives.yaml]: faulty rebase, fixed now!

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33030711).

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33031726).

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33031726).

**Review by albanD — APPROVED** [review]
Thanks for the cleanup of the rebase

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33031726).

**Pull request closed without merging**